packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/OutputLibrary.cs

```diff
@@ -1,84 +1,60 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
 using System.Collections.Generic;
-using Microsoft.Generator.CSharp.Input;
+using Microsoft.CodeAnalysis;
+using System.IO;
 using Microsoft.Generator.CSharp.Providers;
 
 namespace Microsoft.Generator.CSharp
 {
     public class OutputLibrary
     {
-        private IReadOnlyList<EnumProvider>? _enums;
-        private IReadOnlyList<ModelProvider>? _models;
-        private IReadOnlyList<ClientProvider>? _clients;
-
         public OutputLibrary()
         {
-            EnumMappings = new Dictionary<InputEnumType, EnumProvider>();
-            ModelMappings = new Dictionary<InputModelType, ModelProvider>();
-
-            _allModels = new(InitializeAllModels);
         }
 
-        private readonly Lazy<(EnumProvider[] Enums, ModelProvider[] Models)> _allModels;
+        private IReadOnlyList<TypeProvider>? _outputTypes;
+        public IReadOnlyList<TypeProvider> OutputTypes => _outputTypes ??= BuildOutputTypes();
 
-        private (EnumProvider[] Enums, ModelProvider[] Models) InitializeAllModels()
+        protected virtual IReadOnlyList<TypeProvider> BuildOutputTypes()
```

Comment:
Since this is on the OutputLibrary already, should this just be BuildTypes?

Reply:
I chose `BuiltTypes` and also renamed the property to `Types`